odd-collector-sdk/odd_collector_sdk/secrets/aws/ssm_parameter_store.py

```diff
@@ -15,8 +18,12 @@ def __init__(self, **kwargs) -> None:
         collector_plugins_prefix = kwargs.get(
             "collector_plugins_prefix", "/odd/collector_config/plugins"
         )
+        aws_region = kwargs.get("region_name")
 
-        self._region_name = kwargs.get("region_name")
+        self._region_name = os.getenv(
+            "AWS_REGION",
+            self._dynamically_get_aws_region(aws_region),
+        )
```

Comment:
this is a little bit confusing tbh. So what is the correct order? I would actually move all this staff under `get_aws_region` method.
```
self._region_name = self._get_aws_region()
...

def def _get_aws_region(self):
  region = os.getenv("AWS_REGION") or kwargs.get("region_name")
  if region is not None:
        return region

  # otherwise get it from somewhere else: yaml, IMDSv2, etc.
  try:
     # code
```
Pretty straightforward  approach that shows what is the most prioritized value and their order.

Reply:
Already resolved in #35